File: app/src/main/java/com/elegion/android/Lifecycler.java
package com.elegion.android;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;
import android.support.annotation.NonNull;

/**
 * @author Mikhail Barannikov
 */
public class Lifecycler implements Application.ActivityLifecycleCallbacks {

    private static int sStarted;
    private static int sStopped;

    public static void register(@NonNull Application app) {
        app.registerActivityLifecycleCallbacks(new Lifecycler());
    }

    public static boolean isApplicationVisible() {
        return sStarted > sStopped;
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
        // Do nothing.
    }

    @Override
    public void onActivityStarted(Activity activity) {
        sStarted++;
    }

    @Override
    public void onActivityResumed(Activity activity) {
        // Do nothing.
    }

    @Override
    public void onActivityPaused(Activity activity) {
        // Do nothing.
    }

    @Override
    public void onActivityStopped(Activity activity) {
        sStopped++;
    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
        // Do nothing.
    }

    @Override
    public void onActivityDestroyed(Activity activity) {
        // Do nothing.
    }

}

File: app/src/main/kotlin/com/bankex/wallet/Lifecycler.kt
package com.bankex.wallet

import android.app.Activity
import android.app.Application
import android.os.Bundle

open class Lifecycler : Application.ActivityLifecycleCallbacks {

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
        // Do nothing.
    }

    override fun onActivityStarted(activity: Activity) {
        sStarted++
    }

    override fun onActivityResumed(activity: Activity) {
        // Do nothing.
    }

    override fun onActivityPaused(activity: Activity) {
        // Do nothing.
    }

    override fun onActivityStopped(activity: Activity) {
        sStopped++
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {
        // Do nothing.
    }

    override fun onActivityDestroyed(activity: Activity) {
        // Do nothing.
    }

    companion object {
        private var sStarted: Int = 0
        private var sStopped: Int = 0

        @JvmStatic
        fun register(app: Application) {
            app.registerActivityLifecycleCallbacks(Lifecycler())
        }

        val isApplicationVisible: Boolean
            get() = sStarted > sStopped
    }
}
